executor/update_test.go

```diff
@@ -233,3 +233,82 @@ func (s *testUpdateSuite) TestUpdateMultiDatabaseTable(c *C) {
 	tk.MustExec("create table test2.t(a int, b int generated always  as (a+1) virtual)")
 	tk.MustExec("update t, test2.t set test.t.a=1")
 }
+
+var _ = SerialSuites(&testSuite11{&baseTestSuite{}})
+
+type testSuite11 struct {
+	*baseTestSuite
+}
+
+func (s *testSuite11) TestUpdateClusterIndex(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec(`set @@tidb_enable_clustered_index=true`)
+	tk.MustExec(`use test`)
+
+	tk.MustExec(`drop table if exists t`)
+	tk.MustExec(`create table t(id varchar(200) primary key, v int)`)
+	tk.MustExec(`insert into t(id, v) values ('abc', 233)`)
+	tk.MustQuery(`select id, v from t where id = 'abc'`).Check(testkit.Rows("abc 233"))
+	tk.MustExec(`update t set id = 'dfg' where id = 'abc'`)
+	tk.MustQuery(`select * from t`).Check(testkit.Rows("dfg 233"))
+	tk.MustExec(`update t set id = 'aaa', v = 333 where id = 'dfg'`)
+	tk.MustQuery(`select * from t where id = 'aaa'`).Check(testkit.Rows("aaa 333"))
+	tk.MustExec(`update t set v = 222 where id = 'aaa'`)
+	tk.MustQuery(`select * from t where id = 'aaa'`).Check(testkit.Rows("aaa 222"))
+	tk.MustExec(`insert into t(id, v) values ('bbb', 111)`)
+	c.Assert(tk.ExecToErr(`update t set id = 'bbb' where id = 'aaa'`).Error(), Equals, `[kv:1062]Duplicate entry '{bbb}' for key 'PRIMARY'`)
```

Comment:
How about using `tk.MustGetErrCode("xx", errno.ErrDupEntry)` to improve the flexibility for the future changes to error message?